File: pset6/mario.py
from cs50 import get_int

# Main function which calls the get input function and prints the piramid


def main():

    height = get_user_input()

    # Prints the piramid on the basis of the number of rows the user has given as input in the height variable
    for i in range(height):
        for j in range(i + 1, height):
            print(" ", end="")
        for k in range(i + 1):
            print("#", end="")
        for l in range(2):
            print(" ", end="")
        for m in range(i + 1):
            print("#", end="")
        print()

# Promts the user for input until he gives a number between 1 and 8 and return that number


def get_user_input():
    while True:
        i = get_int("Height: ")
        if i >= 1 and i <= 8:
            return i


# Calls the main function
if __name__ == "__main__":
    main()

File: pset1/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int number;
// Promts the user for a number between 1 and 8, until they input one
    do
    {
        number = get_int("Enter number between 1 and 8: ");
    }
    while (number < 1 || number > 8);
    // Creates the rows
    for (int i = 0; i < number; i++)
    {
        // Prints the spaces
        for (int j = number; j > i + 1; j--)
        {
            printf(" ");
        }
        //Prints the hashes
        for (int k = 0; k < i + 1; k++)
        {
            printf("#");
        }
        // Prints two spaces between the hashes
        for (int m = 0; m < 2; m++)
        {
            printf(" ");
        }
        // Prints the second hashes
        for (int k = 0; k < i + 1; k++)
        {
            printf("#");
        }
        printf("\n");
    }
}
